crates/bevy_log/src/lib.rs

```diff
@@ -130,51 +198,98 @@ impl Plugin for LogPlugin {
+                                if let Some(fields) =
+                                    span.extensions().get::<FormattedFields<DefaultFields>>()
+                                {
+                                    format!(
+                                        "{}: {}",
+                                        span.metadata().name(),
+                                        fields.fields.as_str()
+                                    )
+                                } else {
+                                    span.metadata().name().into()
+                                }
+                            }
+                        }))
+                        .build();
+                    app.world.insert_non_send_resource(guard);
+                    chrome_layer
+                };
 
-            #[cfg(feature = "tracing-tracy")]
-            let tracy_layer = tracing_tracy::TracyLayer::new();
+                #[cfg(feature = "tracing-tracy")]
+                let tracy_layer = tracing_tracy::TracyLayer::new();
 
-            let fmt_layer = tracing_subscriber::fmt::Layer::default().with_writer(std::io::stderr);
+                let fmt_layer =
+                    tracing_subscriber::fmt::Layer::default().with_writer(std::io::stderr);
 
-            // bevy_render::renderer logs a `tracy.frame_mark` event every frame
-            // at Level::INFO. Formatted logs should omit it.
-            #[cfg(feature = "tracing-tracy")]
-            let fmt_layer =
-                fmt_layer.with_filter(tracing_subscriber::filter::FilterFn::new(|meta| {
-                    meta.fields().field("tracy.frame_mark").is_none()
-                }));
+                // bevy_render::renderer logs a `tracy.frame_mark` event every frame
+                // at Level::INFO. Formatted logs should omit it.
+                #[cfg(feature = "tracing-tracy")]
+                let fmt_layer =
+                    fmt_layer.with_filter(tracing_subscriber::filter::FilterFn::new(|meta| {
+                        meta.fields().field("tracy.frame_mark").is_none()
+                    }));
 
-            let subscriber = subscriber.with(fmt_layer);
+                let subscriber = subscriber.with(fmt_layer);
+
+                #[cfg(feature = "tracing-chrome")]
+                let subscriber = subscriber.with(chrome_layer);
+                #[cfg(feature = "tracing-tracy")]
+                let subscriber = subscriber.with(tracy_layer);
+                subscriber
+            };
 
-            #[cfg(feature = "tracing-chrome")]
-            let subscriber = subscriber.with(chrome_layer);
-            #[cfg(feature = "tracing-tracy")]
-            let subscriber = subscriber.with(tracy_layer);
+            let file_appender_layer = if let Some(settings) = &self.file_appender_settings {
+                if settings.use_panic_hook {
+                    let old_handler = std::panic::take_hook();
+                    std::panic::set_hook(Box::new(|panic_info| {
```

Comment:
```suggestion
                    std::panic::set_hook(Box::new(move |panic_info| {
```
forgot the `move` sorry